ShellCheck/Parser.hs

```diff
@@ -133,6 +133,38 @@ almostSpace =
         char c
         return ' '
 
+withNextId :: Monad m => ParsecT s UserState (SCBase m) (Id -> b) -> ParsecT s UserState (SCBase m) b
+withNextId p = do
+    start <- getPosition
+    id <- createId
+    setStartPos id start
+    fn <- p
+    let t = fn id
+    end <- getPosition
+    setEndPos id end
```

Comment:
Many of the parsers will also parse and discard trailing spaces, and the end position would then be after all the whitespace. Is there a way around that?